java/adapter/jdbc/src/main/java/org/apache/arrow/adapter/jdbc/JdbcToArrowUtils.java

```diff
@@ -0,0 +1,343 @@
+                    case Types.DECIMAL:
+                        DecimalVector decimalVector = (DecimalVector)root.getVector(columnName);
+                        decimalVector.setSafe(rowCount, rs.getBigDecimal(i));
+                        decimalVector.setValueCount(rowCount + 1);
+                        break;
+                    case Types.REAL:
+                    case Types.FLOAT:
+                        Float4Vector float4Vector = (Float4Vector)root.getVector(columnName);
+                        float4Vector.setSafe(rowCount, rs.getFloat(i));
+                        float4Vector.setValueCount(rowCount + 1);
+                        break;
+                    case Types.DOUBLE:
+                        Float8Vector float8Vector = (Float8Vector)root.getVector(columnName);
+                        float8Vector.setSafe(rowCount, rs.getDouble(i));
+                        float8Vector.setValueCount(rowCount + 1);
+                        break;
+                    case Types.CHAR:
+                    case Types.NCHAR:
+                    case Types.VARCHAR:
+                    case Types.NVARCHAR:
+                    case Types.LONGVARCHAR:
+                    case Types.LONGNVARCHAR:
+                        VarCharVector varcharVector = (VarCharVector)root.getVector(columnName);
+                        String value = rs.getString(i) != null ? rs.getString(i) : "";
+                        varcharVector.setIndexDefined(rowCount);
+                        varcharVector.setValueLengthSafe(rowCount, value.length());
+                        varcharVector.setSafe(rowCount, value.getBytes(Charset.forName("UTF-8")), 0, value.length());
+                        varcharVector.setValueCount(rowCount + 1);
+                        break;
+                    case Types.DATE:
+                        Date date = rs.getDate(i);
+                        DateMilliVector dateMilliVector = (DateMilliVector) root.getVector(columnName);
+                        dateMilliVector.setValueCount(rowCount + 1);
+                        if (date != null) {
+                            dateMilliVector.setSafe(rowCount, rs.getDate(i).getTime());
+                        } else {
+                            dateMilliVector.setNull(rowCount);
+                        }
+                        break;
+                    case Types.TIME:
+                        Time time = rs.getTime(i);
+                        TimeMilliVector timeMilliVector = (TimeMilliVector)root.getVector(columnName);
+                        timeMilliVector.setValueCount(rowCount + 1);
+                        if (time != null) {
+                            timeMilliVector.setSafe(rowCount, (int) rs.getTime(i).getTime());
+                        } else {
+                            timeMilliVector.setNull(rowCount);
+                        }
+
+                        break;
+                    case Types.TIMESTAMP:
+                        // timezone is null
```

Comment:
is it?

Reply:
Yes, I agree this needs to be handled. I will give some more thought.